### DISCUSSION TITLE
wal2json compatibility with postgres versions

### DISCUSSION THREAD

**Discussion by farcepecunpay** [eulerto/wal2json#217] [Q&A] [answered]
Before migrating the database we want to start using wal2json in a postgres 9.4. Is there compatibility or is there any limitation?

**Comment by eulerto** [comment, marked as answer]
wal2json works with Postgres 9.4. However, there are features only available in later Postgres versions (such as TRUNCATE and messages). That's because the logical decoding API does not provide for it.

[Postgres 9.4 is not supported since Feb, 2020](https://www.postgresql.org/support/versioning/). You should upgrade as soon as possible.
